cmd/notation/sign.go

```diff
@@ -112,8 +115,16 @@ func runSign(command *cobra.Command, cmdOpts *signOpts) error {
 	_, err = notation.Sign(ctx, signer, sigRepo, opts)
 	if err != nil {
 		var errorPushSignatureFailed notation.ErrorPushSignatureFailed
-		if errors.As(err, &errorPushSignatureFailed) && !ociImageManifest {
-			return fmt.Errorf("%v. Possible reason: target registry does not support OCI artifact manifest. Try removing the flag `--signature-manifest artifact` to store signatures using OCI image manifest", err)
+		if errors.As(err, &errorPushSignatureFailed) {
+			if !ociImageManifest {
+				return fmt.Errorf("%v. Possible reason: target registry does not support OCI artifact manifest. Try removing the flag `--signature-manifest artifact` to store signatures using OCI image manifest", err)
+			}
+			if strings.Contains(err.Error(), referrersTagSchemaDeleteError) {
+				// failed to delete dangling referrers index
+				fmt.Fprintln(os.Stderr, "Warning: failed to delete dangling referrers index, since this is an OCI v1.0 compliant registry and the deletion API is disabled by the registry.")
```

Comment:
> since this is an OCI v1.0 compliant registry and the deletion API is disabled by the registry

This might not be accurate, except the case that 405(deletion API is disabled), the error might also come from  OCI v1.1 registry with 403(client token don't have access to delete).

Reply:
Code updated.